cpp/src/Glacier2/FilterManager.cpp

```diff
@@ -154,9 +154,9 @@ Glacier2::FilterManager::FilterManager(shared_ptr<Instance> instance, shared_ptr
         auto adapter = _instance->serverObjectAdapter();
         if(adapter)
         {
-            _categoriesPrx = Ice::uncheckedCast<Glacier2::StringSetPrx>(adapter->addWithUUID(_categories));
-            _adapterIdsPrx = Ice::uncheckedCast<Glacier2::StringSetPrx>(adapter->addWithUUID(_adapters));
-            _identitiesPrx = Ice::uncheckedCast<Glacier2::IdentitySetPrx>(adapter->addWithUUID(_identities));
+            _categoriesPrx = Glacier2::StringSetPrx(adapter->addWithUUID(_categories));
```

Comment:
Looking at this code, the proxies are either all null or all not-null.

Is adapter ever null? Does it make sense to create this FilterManager when adapter is null?

Reply:
It is null when [Glacier2.Server.Endpoints](https://doc.zeroc.com/ice/3.7/property-reference/glacier2#id-.Glacier2.*v3.7-Glacier2.Server.AdapterProperty) is not set. Which is only required for callbacks.

Reply:
Looking at the doc link, it's not clear to me why we create this FilterManager object when adapter is null.

From the doc you quoted, it sounds like in this situation, we provide a null session control. Not a session control that returns null proxies.

On a side note, this is a really bad Slice API. Slice operations are supposed to be coarse, not property-like.

Reply:
The filters can be configured statically using properties and the FilterManager is still used.

The relation of session control with the filter manager is to allow dynamic configuration.